service/controller/resource/awstaglist/resource.go

```diff
@@ -0,0 +1,65 @@
+package awstaglist
+
+import (
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+	"github.com/giantswarm/aws-tag-operator/client/aws"
+	"github.com/giantswarm/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+)
+
+const (
+	Name = "aws_tag_list"
+)
+
+type Config struct {
+	K8sClient  k8sclient.Interface
+	Logger     micrologger.Logger
+	AWSClients aws.Interface
+}
+
+type Resource struct {
+	logger    micrologger.Logger
+	k8sClient k8sclient.Interface
+
+	AWSClients aws.Interface
```

Comment:
```suggestion
	awsClients aws.Interface
```

Nit but I'd name it like that and have it first.

Reply:
yeah I had bad times to understand upper/lower case, need to read fmt

Reply:
It's because in Go upper case names are exported and can be used by other packages.

So the config vars are exported for passing in by the calling package. In the constructor these are added to the resource struct and can be kept internal.